app/src/androidTest/java/org/mozilla/tv/firefox/ui/robots/NavigationOverlayRobot.kt

```diff
@@ -34,6 +35,7 @@ class NavigationOverlayRobot {
     fun reload() = reloadButton().click()
     fun toggleTurbo() = turboButton().click()
     fun openSettings() = settingsButton().click()
+    fun toggleDesktopMode() = desktopModeButton().click()
```

Comment:
Suggest breaking this into `turnDesktopModeOn` and `turnDesktopModeOff`, and having each assert the proper state before clicking.

Reply:
Is this necessary though since I already assert the state with `assertCanTurnDesktopModeOn` in the test before clicking?

Reply:
I wouldn't say necessary, but I think it's a good practice.  I don't think that needs to be two separate lines, and I feel that this way reads better.  But it's just a suggestion, if you feel strongly opposed we can leave it as is.

Reply:
Ok, so you're saying call `assertCanTurnDesktopModeOn` in `turnDesktopModeOn/Off` instead of in the actual test?